include/nuttx/compiler.h

```diff
@@ -144,6 +144,7 @@
 #  define noinline_function __attribute__ ((noinline))
 
 #  define printflike(a, b) __attribute__((__format__ (__printf__, a, b)))
+#  define sysloglike(a, b) __attribute__((__format__ (__printf__, a, b)))
```

Comment:
should we reuse printflike macro?

Reply:
i separated them because syslog is not quite printf.
the implementations in gcc/clang might or might not need to distinguish them. (i guess not. but i dunno.)
anyway the separation makes the intention clear.